packages/origin-247-certificate/src/evented-certificate/offchain-certificate.service.ts

```diff
@@ -142,4 +98,12 @@ export class OffchainCertificateFacade<T = null> {
     private generateInternalCertificateId(): number {
         return Math.floor(Math.random() * Number.MAX_SAFE_INTEGER);
     }
+
+    private async propagate(
+        event: ICertificateEvent,
+        command: CertificateCommandEntity
+    ): Promise<void> {
+        this.eventBus.publish(event);
+        this.certEventRepo.save(event, command.id);
```

Comment:
please add `await` here. Also add `await` everywhere propagate is used.

Reply:
We are really close to finishing this part.

Reply:
added